Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -171,11 +160,47 @@ void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
 void UPSWorldSubsystem::Deinitialize()
 {
 	Super::Deinitialize();
-	PSHUDComponentInternal = nullptr;
-	PSCurrentSpotComponentInternal = nullptr;
-	SaveGameDataInternal = nullptr;
-	ProgressionSettingsDataInternal.Empty();
-	StarDynamicProgressMaterial = nullptr;
+}
+
+// Invoked after a game feature plugin is unloaded
+void UPSWorldSubsystem::OnGameFeatureUnloading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	DestroyStarActors();
+	IGameFeatureStateChangeObserver::OnGameFeatureUnloading(GameFeatureData, PluginURL);
+}
+
+// Invoked in the early stages of the game feature plugin loading phase
+void UPSWorldSubsystem::OnGameFeatureLoading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	const FString ProgressionPlugin = TEXT("/ProgressionSystem/ProgressionSystem");
+	if (PluginURL.Contains(ProgressionPlugin))
+	{
+		WolrdSubSystemInitialize();
+	}
+
+	IGameFeatureStateChangeObserver::OnGameFeatureLoading(GameFeatureData, PluginURL);
+}
+
+// Is called to initialize the world subsystem
+void UPSWorldSubsystem::WolrdSubSystemInitialize()
```

Comment:
Since `WolrdSubSystemInitialize` is called from two places, that is expected, BeginPlay and OnLoading, so you should filter second call, to avoid double initialization. It will be called twice likely on very first start of the game in editor.

Please add `if` early return to guarantee you only initialize once. Don't add new cached properties, but check it by other known conditions. Like if observer is already added, then it was initialize before, then early return etc.

Reply:
We discussed that we will migrate to new logic with automatic MGF  load/unload so this will be no longer a case. I propose we skip this for now